compiler/modules/bir/verify.bal

```diff
@@ -65,13 +65,223 @@ class VerifyContext {
     }
 }
 
+// Marks which registers are initialized from a given path into a block
+type RegFlow record {|
+    int? origin; // origin == () iff entry
+    boolean[] regs;
+|};
+
 public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode code) returns Error? {
     VerifyContext cx = new(mod, defn);
-    foreach BasicBlock b in code.blocks {
+    Position funcPos = defn.position;
+    BasicBlock[] blocks = code.blocks;
+    int numBlocks = blocks.length();
+    if numBlocks == 0 {
+        return cx.invalidErr("no basic blocks for function", funcPos);
+    }
+    // Checks are done in two DFS rounds. During first round, forward in degree is calculated for each block.
+    int[] fwdInDegree = [];
```

Comment:
Is "degree" the standard name for number of incoming edges?

Reply:
Yes, 'in-degree' is a [standard term](https://en.wikipedia.org/wiki/Glossary_of_graph_theory#D).